src/main/java/com/github/checkstyle/regression/Main.java

```diff
@@ -34,6 +72,110 @@ private Main() {
      * @throws Exception execute failure
      */
     public static void main(String[] args) throws Exception {
-        // empty for now
+        final Options options = createOptions();
+        final CommandLineParser parser = new DefaultParser();
+        final HelpFormatter formatter = new HelpFormatter();
+        final CommandLine cmd;
+
+        try {
+            cmd = parser.parse(options, args);
+            final String repoPath = cmd.getOptionValue(OPT_LOCAL_GIT_REPO);
+            if (!existAndIsDirectory(repoPath)) {
+                throw new IllegalArgumentException(
+                        "path of local git repo must exist and be a directory");
+            }
+            final String branch = cmd.getOptionValue(OPT_PATCH_BRANCH);
+            final boolean stopAfterGenerateConfig = cmd.hasOption(OPT_STOP_AFTER_GENERATE_CONFIG);
+
+            final File config = generateConfig(repoPath, branch);
```

Comment:
> What is runRegression? Do you mean create a new method with name runRegression?

yes, and move these liens into it.
I want separation between validating input and running of regression.
Similar to https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/Main.java#L381 and https://github.com/checkstyle/checkstyle/blob/master/src/main/java/com/puppycrawl/tools/checkstyle/Main.java#L287 .